RobotNavigation/MazeConfigReader.cs

```diff
@@ -12,19 +12,19 @@ public class MazeConfigReader
     {
         private StreamReader _file;
 
-        private Maze.Cell[] _targets;           // TL - see above
+        private Position[] _targets;           // TL - see above
```

Comment:
Why do you need to store this?  Aren't the targets stored inside the Maze object?  Can't you just add them one at a time using a SetTarget method in the maze?